### PR TITLE
Create doc how to setup rn feature

### PR DESCRIPTION
**Pull request by lukewalczak**
Fixes: #1133

### CODE REVIEW

File: docs/common/create_new_rn_feature.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+│           │── scenes/            - folder with custom package screens
+│           │── index.js           - file which is exporting CustomStandalonePackage
+│           └── package.json
+```
+
+## appRegistry
+
+### RootComponent
+
+That file is representing the context providers which are passing data through the component tree for your application without having to pass props down manually at every level.
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "This".

File: docs/common/create_new_rn_feature.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    accessToken={this.props.accessToken || null}
+    bookingId={this.props.bookingId || null}
+    simpleToken={this.props.simpleToken || null}
+  >
+    {this.props.children}
+  </AuthContext.Provider>
+</Dimensions.Provider>
+```
+
+_For more details check [Context documentation]('https://reactjs.org/docs/context.html')_
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: The link is not working, also add a "." at the end.

File: docs/common/create_new_rn_feature.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+# Create new RN feature from the scratch
+
+Within the `app` folder create new folder dedicated for your package eg. `custom` and then follow the structure below:
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "a new package" and "e.g.".

File: docs/common/create_new_rn_feature.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    simpleToken={this.props.simpleToken || null}
+  >
+    {this.props.children}
+  </AuthContext.Provider>
+</Dimensions.Provider>
+```
+
+_For more details check [Context documentation]('https://reactjs.org/docs/context.html')_
+
+### CustomStandalonePackage
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: When I read this:

1. I do not know how the `CustomStandalonePackage` should look like
2. I do not know why do I need it or what is it used for

File: docs/common/create_new_rn_feature.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+```js
+export default WithNativeNavigation(CustomStandalonePackage, 'Custom');
+```
+
+## navigation
+
+### CustomStack
+
+Within `navigation` folder create your `StackNavigator` which provides a way for your app to transition between screens where each new screen is placed on top of a stack.
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "the navigation" and "on top of the stack".

File: docs/common/create_new_rn_feature.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+The example stack navigator should look like:
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "The stack navigator example"

File: docs/common/create_new_rn_feature.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  },
+  {
+    ...StackNavigatorOptions,
+    initialRouteName: 'Custom',
+    mode: 'modal'
+  }
+);
+```
+
+_For more details check [React Navigation documentation]('https://reactnavigation.org/docs/en/stack-navigator.html')_
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: Link not working and the sentence should end with dot.

File: docs/common/create_new_rn_feature.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+The example stack navigator should look like:
+
+```js
+export default StackNavigator(
+  {
+    Custom: {
+      screen: CustomScreen,
+      navigationOptions: {
+        headerLeft: null,
+        ...createTransparentHeaderStyle(RNDimensions.get('screen'))
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: What is RNDimensions? Maybe add the import code for it? Is it the `Dimensions` from `react-native` ?

File: docs/common/create_new_rn_feature.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+);
+```
+
+_For more details check [React Navigation documentation]('https://reactnavigation.org/docs/en/stack-navigator.html')_
+
+## scenes
+
+### CustomScreen
+
+Screen which is on the top of the navigation stack. Within you can define your component, where the only thing that's required is a `render` function which returns some JSX to render.
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "This is the screen..."

Comment (ferrannp) [reply docs/common/create_new_rn_feature.md]: You can just delete this part "which returns some JSX to render.".

File: docs/common/create_new_rn_feature.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+```js
+import * as React from 'react';
+import { Text, View } from 'react-native';
+
+export default class CustomScreen extends Component {
+  render() {
+    return (
+      <View>
+        <Text>This is custom screen</Text>
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "This is a custom screen".

File: docs/common/create_new_rn_feature.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+      </View>
+    );
+  }
+}
+```
+
+## package.json
+
+`Package.json` file holds various metadata relevant to the project and handle the project's dependencies.
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: The `package.json`

File: docs/common/create_new_rn_feature.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+    "@kiwicom/mobile-localization": "^0",
+    "@kiwicom/mobile-navigation": "^0",
+    "@kiwicom/mobile-orbit": "^0",
+    "@kiwicom/mobile-relay": "^0",
+    "@kiwicom/mobile-shared": "^0"
+  }
+}
+```
+
+_For more details check [documentation]('https://docs.npmjs.com/files/package.json')_
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "check the" + link not working + dot at the end.

File: docs/common/create_new_rn_feature.md [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+
+Entry point for your package, where you have to export your standalone package:
+
+```js
+import CustomStandalonePackage from './src/appRegistry/CustomStandalonePackage';
+
+export { CustomStandalonePackage };
+```
+
+## Using package in native
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "Using the".

File: docs/common/create_new_rn_feature.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+When your package is ready to use in native side you have to register it in [`native.js`]('https://github.com/kiwicom/mobile/blob/master/app/native.js') file:
```
Comment (ferrannp) [inline docs/common/create_new_rn_feature.md]: "in the native side".

File: docs/common/create_new_rn_feature.md (hunk 8 above)
Comment (lukewalczak) [reply docs/common/create_new_rn_feature.md]: I just copy paste it from other file. It's not even needed there - going to remove it.

**Pull request merged**